app/components/ui/Navigation.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FormEvent, useState } from 'react';
+import { Link, useHistory, useLocation } from 'react-router-dom';
+import cn from 'classnames';
+import whiteLabel from '../../utils/whitelabel';
+import styles from './Navigation.module.scss';
+
+const {
+  appImages: {
+    logoSmall,
+  },
+  logoLinkDestination,
+  navLogoStyle,
+  showMobileNav,
+  siteNavStyle,
+  siteUrl,
+  title,
+} = whiteLabel;
+
+export const Navigation = ({ showSearch, toggleHamburgerMenu }: {
+  showSearch: boolean;
+  toggleHamburgerMenu: () => void;
+}) => {
+  const [query, setQuery] = useState('');
```

Comment:
I think we may want to do a bit of extra QA in staging for this, since I remember this being very complicated, since there's a very unintuitive sequence of how this is initialized, updated, and kept in sync across different components, especially due to part of this living in the browser history and URL location.

But from what I'm reading in your code changes, I think this looks about right, other than it looking like we previously didn't actually initialize `this.state.query`, meaning that it was probably `undefined`. I definitely prefer your way here, setting it to an explicit value, but I'd just want to be careful about it.